### ISSUE TITLE
[BUG] PDFs not updated

### ISSUE THREAD

**Issue by ms609** [revbayes/revbayes#581]
**Describe the bug**

PDFs not updated before proposal when conducting MCMC with two morphological datasets

```
Error:       Executing Scaling(tree_length): PDFs not up-to-date before proposal!
    m_morph[2]: -1556.4265621 != -1653.7328371    diff = 97.306275003
    m_morph[1]: -13984.554612 != -13903.385021    diff = -81.16959119
```
**To Reproduce**
- Download [data.txt](https://github.com/user-attachments/files/17519572/data.txt) and [data2.txt](https://github.com/user-attachments/files/17637137/data2.txt)

- Run the following:

```rb
seed(0)
setOption("debugMCMC", 1)
moves = VectorMoves()

chars <- readDiscreteCharacterData("data.txt")
taxa <- chars.names()
outgroup <- clade(taxa[1])
nTaxa <- chars.size()
nEdge <- 2 * nTaxa - 3

topology ~ dnUniformTopology(taxa, outgroup = outgroup)
moves.append( mvNNI(topology, weight = nEdge) )

tree_length ~ dnGamma(shape = 2, rate = 2 / 3)
moves.append( mvScale(tree_length ) )

rel_br_lengths ~ dnDirichlet( rep(1.0, nEdge) )
br_lengths := rel_br_lengths * tree_length
phylogeny := treeAssembly(topology, br_lengths)

m_morph[1] ~ dnPhyloCTMC(
  tree = phylogeny,
  Q = fnJC(4),
  type = "Standard",
  coding = "variable"
)
m_morph[1].clamp(chars)

m_morph[2] ~ dnPhyloCTMC(tree = phylogeny, Q = fnJC(2), type = "Standard")
condChars <- readDiscreteCharacterData("data2.txt")
m_morph[2].clamp(condChars)

monitors[1] = mnScreen(printgen = 1, tree_length, posterior = FALSE, prior = FALSE)
mymc3 = mcmc(model(phylogeny), monitors, moves, nruns = 2, moveschedule = "single")

mymc3.run(generations = 1000)
```

**Computer info**
Latest RevBayes build from debug-mcmc branch

**Comment by davidcerny** [comment]
Possibly a duplicate of #567, having to do with `treeAssembly` not properly restoring branch lengths...

**Comment by ms609** [comment]
This does indeed seem to be fixed by #549 – worth verifying once that PR is merged.

**Comment by davidcerny** [comment]
Just checked; after merging #549, the issue indeed no longer occurs on the current tip of development ([a11fcaf](https://github.com/revbayes/revbayes/commit/a11fcaf11d866b9cd5b34398eb9ea15b36b1cd80)).
